code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm

```diff
@@ -140,14 +240,31 @@
 	name = "Carnage"
 	action_icon_state = "carnage"
 	mechanics_text = "For a while your attacks drain blood and heal you. During Feast you also heal nearby allies."
+	use_state_flags = XACT_TARGET_SELF|XACT_IGNORE_SELECTED_ABILITY
 	cooldown_timer = 40 SECONDS
 	plasma_cost = 0
+	keybind_signal = COMSIG_XENOABILITY_CARNAGE
+	keybind_flags = XACT_KEYBIND_USE_ABILITY
 
-/datum/action/xeno_action/activable/carnage/use_ability()
-	var/mob/living/carbon/xenomorph/X = owner
-	X.apply_status_effect(/datum/status_effect/xeno_carnage, 20 SECONDS, X.xeno_caste.carnage_plasma_gain)
-	succeed_activate()
+/datum/action/xeno_action/activable/carnage/use_ability(atom/A)
+	. = ..()
+	var/mob/living/carbon/xenomorph/owner_xeno = owner
+	owner_xeno.apply_status_effect(/datum/status_effect/xeno_carnage, 20 SECONDS, owner_xeno.xeno_caste.carnage_plasma_gain)
 	add_cooldown()
+	succeed_activate()
+
+/datum/action/xeno_action/activable/carnage/ai_should_use(atom/target)
+	var/mob/living/carbon/xenomorph/owner_xeno = owner
+	if(owner_xeno.plasma_stored > owner_xeno.xeno_caste.plasma_max * 0.8)
+		// ability is primarily used for plasma gain, so should be used even at full health
+		if(owner_xeno.health > owner_xeno.maxHealth * 0.9)
+			return FALSE
+		return FALSE
+	// nothing gained by slashing allies
+	if(target.get_xeno_hivenumber() == owner_xeno.get_xeno_hivenumber())
+		return FALSE
+
+	return can_use_ability(target) ? TRUE : ..()
```

Comment:
```suggestion
	return can_use_ability(target)
```

..() just return FALSE all the time. I know they are still some exemple that do that, but it's just stupid and it's code i still didn't remove

Reply:
changed